server/routerlicious/packages/lambdas-driver/src/document-router/documentContext.ts

```diff
@@ -79,15 +81,44 @@ export class DocumentContext extends EventEmitter implements IContext {
 		// Assert offset is between the current tail and head
 		const offset = message.offset;
 
-		assert(
-			offset > this.tail.offset && offset <= this.head.offset,
-			`${offset} > ${this.tail.offset} && ${offset} <= ${this.head.offset} ` +
-				`(${message.topic}, ${message.partition}, ${this.routingKey.tenantId}/${this.routingKey.documentId})`,
-		);
-
-		// Update the tail and broadcast the checkpoint
-		this.tailInternal = message;
-		this.emit("checkpoint", restartOnCheckpointFailure);
+		try {
+			assert(
+				offset > this.tail.offset && offset <= this.head.offset,
+				`${offset} > ${this.tail.offset} && ${offset} <= ${this.head.offset} ` +
+					`(${message.topic}, ${message.partition}, ${this.routingKey.tenantId}/${this.routingKey.documentId})`,
+			);
+
+			// Update the tail and broadcast the checkpoint
+			this.tailInternal = message;
+			this.emit("checkpoint", restartOnCheckpointFailure);
+		} catch (error) {
+			// Mark the document as corrupted
+			const documentId = this.routingKey.documentId;
+			const tenantId = this.routingKey.tenantId;
+			// Update the tail
+			this.tailInternal = message;
+			const properties = {
+				...getLumberBaseProperties(documentId, tenantId),
+				messageOffset: message.offset,
+				headOffset: this.head.offset,
+				tailOffset: this.tail.offset,
+				topic: message.topic,
+				partition: message.partition,
+			};
+			Lumberjack.error(
+				`Skipping checkpoint and marking document as corrupted. Message offset is not between the current tail and head offsets.`,
+				properties,
+				error,
+			);
+			// we emit the error last
+			this.error(error, {
+				restart: false,
+				markAsCorrupt: message,
+				skipCheckpoint: true,
```

Comment:
I think skipping a checkpoint may lead to the kafka offset not being committed. Is it necessary to skip the checkpoint for this?